### PR TITLE
truecrypt: fix build with gcc6 patch

### PR DESCRIPTION
**Pull request by ryantm**
The build was failing with gcc 6.4.0; pinning to gcc5 (gcc-5.4.0)
allows the build to complete.

Fixing as part of #28643.

### CODE REVIEW

File: pkgs/applications/misc/truecrypt/default.nix [hunk 1]
```diff
@@ -41,7 +41,7 @@ library, use the 'NOGUI' parameter:
 */
 
 { fetchurl, stdenv, pkgconfig, nasm, fuse, wxGTK, devicemapper,
-  wxGUI ? true
+  wxGUI ? true, gcc
```
Comment (Mic92) [inline pkgs/applications/misc/truecrypt/default.nix]: It is probably better to apply this patch instead: https://git.archlinux.org/svntogit/packages.git/tree/trunk/gcc6.patch?h=packages/truecrypt

**Comment by ryantm** [comment]
@Mic92 Okay, I incorporated your review suggestion. Can you review again, please?

**Comment by ryantm** [comment]
cc @viric

**Comment by Mic92** [comment]
Thanks!

**Comment by Mic92** [comment]
also backported.

**Pull request merged**